banzai/frames.py

```diff
@@ -218,12 +197,21 @@ def __itruediv__(self, other):
             self.primary_hdu.__itruediv__(other)
         return self
 
+    def __getitem__(self, item):
+        return self._hdus[self._hdu_keys[item]]
+
+    def __setitem__(self, key, value):
+        if not key in self._hdu_keys:
+            self._hdus.append(value)
+            i = len(self._hdus) - 1
+            self._hdu_keys[key] = i
+            self._hdu_keys[self._hdus[-1].name] = i
+        else:
+            self._hdus.remove(self._hdu_keys[key])
```

Comment:
I think you want a `del self._hdus[self._hdu_keys[key]]` since you are removing by index. [remove() removes by value](https://www.programiz.com/python-programming/methods/list/remove)